File: python/day14.py
import threading
import time


# Function to run thread 
def say(message, count):
    for i in range(count):
        print(message)



#  main execution
say("hello", 3)

# Create thread object
t1 = threading.Thread(target=say, args=("World", 5))

t1.start()

# Create and start an anonymous (lambda) thread.
t2 = threading.Thread(target=lambda msg: print(msg), args=("I am a new thread...",))
t2.start()

print("Main is waiting....")
t1.join()
t2.join()

print("Main is done.")

File: go-basic/day14.go
package main

import (
	"fmt"
	"time"
)

// A simple function we want to run concurrently
func say(message string, count int) {
	for i := 0; i < count; i++ {
		fmt.Println(message)
	}
}

func main() {
	say("Hello", 3)
	// run function as goroutine
	go say("world", 5)
	// run another function as goroutine
	go func(msg string) {
		fmt.Println(msg)
	}("I am an anonymous goroutine...")

	fmt.Println("Main is waiting...")
	time.Sleep(1 * time.Second)
	fmt.Println("Main is done.")
}
